include/deal.II/lac/petsc_vector_base.h

```diff
@@ -587,108 +601,124 @@ namespace PETScWrappers
      */
-    bool
-    all_zero() const;
+    virtual bool
+    all_zero() const override;
 
     /**
      * Return @p true if the vector has no negative entries, i.e. all entries
      * are zero or positive. This function is used, for example, to check
      * whether refinement indicators are really all positive (or zero).
      */
     bool
-    is_non_negative() const;
+    is_non_negative()
+      const; // TODO - this is not in VectorSpaceVector - should we deprecate?
 
     /**
      * Multiply the entire vector by a fixed factor.
      */
-    VectorBase &
-    operator*=(const PetscScalar factor);
+    virtual VectorBase &
+    operator*=(const PetscScalar factor) override;
 
     /**
      * Divide the entire vector by a fixed factor.
      */
-    VectorBase &
-    operator/=(const PetscScalar factor);
+    virtual VectorBase &
+    operator/=(const PetscScalar factor) override;
 
     /**
      * Add the given vector to the present one.
      */
-    VectorBase &
-    operator+=(const VectorBase &V);
+    virtual VectorBase &
+    operator+=(const VectorSpaceVector<PetscScalar> &V) override;
 
     /**
      * Subtract the given vector from the present one.
      */
-    VectorBase &
-    operator-=(const VectorBase &V);
+    virtual VectorBase &
+    operator-=(const VectorSpaceVector<PetscScalar> &V) override;
+
+    virtual void
+    import(
+      const ReadWriteVector<PetscScalar> &            V,
+      VectorOperation::values                         operation,
+      std::shared_ptr<const CommunicationPatternBase> communication_pattern =
+        std::shared_ptr<const CommunicationPatternBase>()) override;
 
     /**
      * Addition of @p s to all components. Note that @p s is a scalar and not
      * a vector.
      */
-    void
-    add(const PetscScalar s);
+    virtual void
+    add(const PetscScalar s) override;
 
     /**
      * Simple addition of a multiple of a vector, i.e. <tt>*this += a*V</tt>.
      */
-    void
-    add(const PetscScalar a, const VectorBase &V);
+    virtual void
+    add(const PetscScalar a, const VectorSpaceVector<PetscScalar> &V) override;
 
     /**
      * Multiple addition of scaled vectors, i.e. <tt>*this += a*V+b*W</tt>.
      */
-    void
-    add(const PetscScalar a,
-        const VectorBase &V,
-        const PetscScalar b,
-        const VectorBase &W);
+    virtual void
+    add(const PetscScalar                     a,
+        const VectorSpaceVector<PetscScalar> &V,
+        const PetscScalar                     b,
+        const VectorSpaceVector<PetscScalar> &W) override;
 
     /**
      * Scaling and simple vector addition, i.e. <tt>*this = s*(*this)+V</tt>.
      */
     void
-    sadd(const PetscScalar s, const VectorBase &V);
+    sadd(const PetscScalar s, const VectorBase &V); // TODO - not in
+                                                    // VectorSpaceVector. Should
+                                                    // we deprecate?
```

Comment:
do we use this function in any of the linear solver?

Reply:
I'm not sure, but deprecating it is wrong since `LA::D::Vector` provides it.